week2/Q1.py

```diff
@@ -0,0 +1,32 @@
+from btree import BinaryTree
+
+
+def get_ancestors(tree, key):
+	'''Returns ancestors of one key (not including itself).
+
+	Args:
+		tree: BinaryTree, search is conducted on it 
+		key: element of a tree (type not specified), which ancestors algorithm is looking for
+	
+	Returns:
+		list, ancestors of the key, beginning from the closest
+		None, if the key is not present in the tree'''
+	if tree.root != key:
```

Comment:
Optional:
When you have an 'unbalanced' `if`-statement (especially with a return in one branch), it is usually more readable to have the shorter branch first.

Compare:
```python
  if [...]:
    [...]
    [...]
    [...]
    [...]
    [...]
  else:
    return [...]
```

with:

```python
  if not [...]:
    return [...]
  [...]
  [...]
  [...]
  [...]
  [...]
```